Support DragonFly BSD

mneumann opened the pull request:
All tests pass given cmake is configured with
-DMBEDTLS_ROOT_DIR=/usr/local.

Likely, to support the two other BSDs NetBSD and OpenBSD, their respective identifiers would have to be added there as well.

These changes are made under both the "Apache 2.0" and the "GNU Lesser General
Public License 2.1 or later" license terms (dual license).

SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later

bettio reviewed (changes requested):
Looks great, would you mind to use `--signoff` commit option?

bettio commented:
I also suggest rebasing this on top of release-0.6 so DragonFly BSD support can land on latest stable version and adding this to CHANGELOG.Md file (just add `# Added` section for v0.6.2).

UncleGrumpy commented:
One quick question, can you confirm that `os:type/0` returns `{unix,dragonfly}`, and the shell returns `DragonFly` for `uname`, so I can add DragonFly support to #1117?

mneumann commented:
@UncleGrumpy 

See my commit message, or:

    $ erl
    1> os:type().
    {unix,dragonfly}

    $ uname -s
    DragonFly

    $ uname -r
    6.5-DEVELOPMENT

    $ uname -P
    dragonfly:6.6:x86:64

mneumann commented:
Signed off. Rebased against release-0.6 (PR is now against release-0.6). Added ChangeLog entry. Addressed comments on os:type() and uname on DragonFly.
